packages/rancher-vsphere-cpi/charts/templates/_helpers.tpl

```diff
@@ -5,3 +5,14 @@
 {{- "" -}}
 {{- end -}}
 {{- end -}}
+
+{{- /* Return the image's repository and tag corresponding to the cluster's kubernetes version. */ -}}
+{{- define "get_image" -}}
+{{- $kubeVersion := printf "%s.%s" .Capabilities.KubeVersion.Major .Capabilities.KubeVersion.Minor -}}
+{{- if hasKey .Values.cloudControllerManager.image $kubeVersion -}}
```

Comment:
Can we add a check to see if `.Values.cloudControllerManager.image.override.repository` or `.Values.cloudControllerManager.image.override.tag` are set here? If either is set, we should respect a user override over trying to calculate the kubeVersion. If both are set, we shouldn't even try doing the kubeVersion check.

We shouldn't expose the override field in the values.yaml (only accept it if a user provides it), but this would allow us to bake-in a solution in case a user needs to override the image + tag for some reason.

Reply:
I agree the user should have the option to override it, but due to the way this chart works (image tags based on the k8s version line), I think the user should set both a repository and tag for the override to work. If they override the repository only, we would need to guess the tag based on the cluster's k8s version which I don't think is something we should do.